types/readable-stream/index.d.ts

```diff
@@ -4,12 +4,12 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
-/// <reference types="node" />
+/// <reference types="node/v10" />
```

Comment:
Isn't that going to break when installed?

Reply:
To my understanding this results in @types/node v10 installed if @types/readable-stream is the only one asking for node definitions (or all others don't care about the version) which should be fine.

In case someone has additionally @types/node v12.x as dependency the resulting folder structure should be as follows:
`node_modules/@types/node/`  => v12.x definitions
`node_modules/@types/readable-stream/node_modules/@types/node/`=> v10.x definitions
`node_modules/@types/readable-stream/` the readable stream definitions

I put this together in gave this a fast try and it seems to work otherwise I wouldn't create such a PR :grin:. But yes, I still have not the best feeling with this binding.

Any input from typescript/type definiton experts is welcome regarding this!

Reply:
Imo it would be better to entirely decouple this package from node streams and basically copy paste stuff in there.